### PR TITLE
ASoC/SOF/SoundWire: fix suspend-resume on pause with dynamic pipelines

### PR DESCRIPTION
**Pull request by plbossart**
We forgot about this patchset, it's very much needed for the transition to dynamic pipelines with SoundWire.

### CODE REVIEW

**Review by ranj063 — APPROVED** [review]
looks good @plbossart. BTW some of the commit titles have / instead of :. not sure if it is intentional

**Comment by plbossart** [comment]
> looks good @plbossart. BTW some of the commit titles have / instead of :. not sure if it is intentional

yes it was intentional, it's a commit that impacts ASoC and SoundWire subsystems and two maintainers will have to agree

File: drivers/soundwire/intel.c [hunk 1]
```diff
@@ -1023,11 +1001,97 @@ static void *intel_get_sdw_stream(struct snd_soc_dai *dai,
...
+
+		if (dma->suspended)
+			continue;
+
+		if (dma->paused) {
+			dma->suspended = true;
+
+			ret = intel_free_stream(sdw, stream, dai, sdw->instance);
+			if (ret < 0)
+				return ret;
```
Comment (RanderWang) [inline drivers/soundwire/intel.c:1082]: return error or go on releasing other stream and return error ?

**Comment by bardliao** [comment]
@plbossart It looks good to me, but I would like to test it. Could you remine me which topology that has two FEs connect to one BE?

**Comment by plbossart** [comment]
> @plbossart It looks good to me, but I would like to test it. Could you remine me which topology that has two FEs connect to one BE?

https://github.com/thesofproject/sof/pull/4611

File: drivers/soundwire/intel.c (hunk 1 above)
Comment (plbossart) [reply drivers/soundwire/intel.c:1082]: Good question. I think it doesn't matter: if we have an error at this point, it's the firmware that responded with an IPC error, and at this point it's not really helpful to try to free the other streams handled by the DSP.

@bardliao @ranj063 IIRC for other DAIs we also stop on the first error, don't we?

**Review by bardliao — APPROVED** [review]
LGTM and tested on TGL platform

**Pull request merged**